Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/Internal/TenantManagementService.cs

```diff
@@ -444,6 +445,57 @@ public async Task<ServiceManifestRequiredConfigurationEntry[]> GetServiceEnrollm
             return requirements.ToArray();
         }
 
+        /// <inheritdoc/>
+        public async Task AddConfigurationAsync(ITenant tenant, ConfigurationItem[] configurationItems)
+        {
+            if (tenant == null)
+            {
+                throw new ArgumentNullException(nameof(tenant));
+            }
+
+            tenant.EnsureTenantIsOfType(MarainTenantType.Client, MarainTenantType.Delegated, MarainTenantType.Service);
+
+            if (configurationItems == null)
+            {
+                throw new ArgumentNullException(nameof(configurationItems));
+            }
+
+            this.logger.LogDebug(
+                "Add configuration for tenant '{tenantName}' with Id '{tenantId}'",
+                tenant.Name,
+                tenant.Id);
+
+            configurationItems.ValidateAndThrow();
+
+            IEnumerable<KeyValuePair<string, object>> propertiesToAddToTenant = PropertyBagValues.Empty;
+
+            foreach (ConfigurationItem configurationItem in configurationItems)
+            {
+                this.logger.LogDebug(
+                    "Adding configuration entry '{configurationKey}' to tenant '{tenantName}' with Id '{tenantId}'",
+                    configurationItem.Key,
+                    tenant.Name,
+                    tenant.Id);
+
+                propertiesToAddToTenant = configurationItem.AddConfiguration(propertiesToAddToTenant);
+            }
+
+            this.logger.LogDebug(
+                "Updating tenant '{tenantName}' with Id '{tenantId}'",
+                tenant.Name,
+                tenant.Id);
+
+            tenant = await this.tenantStore.UpdateTenantAsync(
+                tenant.Id,
+                propertiesToSetOrAdd: propertiesToAddToTenant)
```

Comment:
Worth noting that as it stands, this code will overwrite existing config (i.e. it's really "add or update", not "add"). I think we should consider either:
- Renaming the command to reflect this; or,
- Blowing up if you try and update a property that's already set; or,
- Adding an extra parameter to say that you're ok with overwriting existing properties, and blowing up if it's not set and you're trying to set an existing property.